File: src/workflow-config/trigger-builder.ts
/**
 * 触发条件构建器
 *
 * 负责构建工作流的触发条件配置。
 */

import type { TriggerConfig, InputConfig } from '../types/workflow-types';

export class TriggerBuilder {
  private _triggers: TriggerConfig = {};

  /**
   * 添加 push 触发条件
   */
  onPush(branches: string[] | string = ['main', 'master']): this {
    this._triggers.push = {
      branches: Array.isArray(branches) ? branches : [branches],
    };
    return this;
  }

  /**
   * 添加 pull_request 触发条件
   */
  onPullRequest(branches: string[] | string = ['main', 'master']): this {
    this._triggers.pull_request = {
      branches: Array.isArray(branches) ? branches : [branches],
    };
    return this;
  }

  /**
   * 添加 release 触发条件
   */
  onRelease(types: string[] | string = ['published']): this {
    this._triggers.release = {
      types: Array.isArray(types) ? types : [types],
    };
    return this;
  }

  /**
   * 添加 workflow_dispatch 触发条件
   */
  onWorkflowDispatch(inputs?: Record<string, InputConfig>): this {
    this._triggers.workflow_dispatch = {};
    if (inputs) {
      this._triggers.workflow_dispatch.inputs = inputs;
    }
    return this;
  }

  /**
   * 添加 schedule 触发条件
   */
  onSchedule(cron: string): this {
    if (!this._triggers.schedule) {
      this._triggers.schedule = [];
    }
    this._triggers.schedule.push({ cron });
    return this;
  }

  /**
   * 构建 TriggerConfig
   */
  build(): TriggerConfig {
    return Object.keys(this._triggers).length > 0 ? { ...this._triggers } : {};
  }

  /**
   * 获取当前触发配置
   */
  getTriggers(): TriggerConfig {
    return { ...this._triggers };
  }
}

File: python/src/workflow_config/trigger_builder.py
"""
触发条件构建器

负责构建工作流的触发条件配置。
"""

from typing import Dict, Optional, Any, List, Union


class TriggerBuilder:
    """触发条件构建器"""
    
    def __init__(self) -> None:
        self._triggers: Dict[str, Any] = {}
    
    def on_push(
        self,
        branches: Union[List[str], str] = None
    ) -> "TriggerBuilder":
        """添加 push 触发条件"""
        if branches is None:
            branches = ["main", "master"]
        self._triggers["push"] = {
            "branches": branches if isinstance(branches, list) else [branches],
        }
        return self
    
    def on_pull_request(
        self,
        branches: Union[List[str], str] = None
    ) -> "TriggerBuilder":
        """添加 pull_request 触发条件"""
        if branches is None:
            branches = ["main", "master"]
        self._triggers["pull_request"] = {
            "branches": branches if isinstance(branches, list) else [branches],
        }
        return self
    
    def on_release(
        self,
        types: Union[List[str], str] = None
    ) -> "TriggerBuilder":
        """添加 release 触发条件"""
        if types is None:
            types = ["published"]
        self._triggers["release"] = {
            "types": types if isinstance(types, list) else [types],
        }
        return self
    
    def on_workflow_dispatch(
        self,
        inputs: Optional[Dict[str, Any]] = None
    ) -> "TriggerBuilder":
        """添加 workflow_dispatch 触发条件"""
        self._triggers["workflow_dispatch"] = {}
        if inputs:
            self._triggers["workflow_dispatch"]["inputs"] = inputs
        return self
    
    def on_schedule(self, cron: str) -> "TriggerBuilder":
        """添加 schedule 触发条件"""
        if "schedule" not in self._triggers:
            self._triggers["schedule"] = []
        self._triggers["schedule"].append({"cron": cron})
        return self
    
    def build(self) -> Dict[str, Any]:
        """构建 TriggerConfig"""
        return dict(self._triggers) if self._triggers else {}
    
    def get_triggers(self) -> Dict[str, Any]:
        """获取当前触发配置"""
        return dict(self._triggers)

